extensions/scrollback-3.3.md

```diff
@@ -0,0 +1,70 @@
+---
+title: IRCv3 scrollback extension
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Evan Magaliff"
+    period: "2017"
+    email: "evan@muffinmedic.net"
+---
+## Description
+This document describes the format of the `scrollback` extension. The `scrollback` extension uses the `chathistory` batch type, adding a client-side command for requesting scrollback content from the server.
+
+The `scrollback` extensions adds an optional `draft/msgid` to the `chathistory` batch reply.
+
+Client side support for the [`batch`][batch] and [`server-time`][server-time] capabilities is required. Support for the [`draft/msgid`](https://github.com/ircv3/ircv3-specifications/pull/285) capability is optional but recommended.
+
+When a client with the above mentioned capabilities requests scrollback content from the server (using the `SCROLLBACK` command outlined below), the server should return to the client a single batch containing a number of desired raw IRC lines equal to the `message_count` parameter specified, ending directly before the given timestamp or with the message directly proceeding the one with the specified `draft/msgid`. The raw IRC lines should be formatted and returned to the client as they were originally, with the addition of the above capability tags.
+
+The `server-time` should be the time at which the message was originally sent, the `batch id` a randomly generated string unique to the entire batch, and the optional `draft/msgid`, if included, the `draft/msgid` originally sent with the message.
+
+### `SCROLLBACK` Command
+Scrollback content can be requested by the client to the server by sending the `SCROLLBACK` command to the server. A `batch` must be returned by the server. If no content exists to return, an empty batch should be returned to avoid the client waiting for a reply. Command support is sent to the client as the RPL_ISUPPORT 005 numeric `:irc.host 005 nick SCROLLBACK :are supported by this server`
+
+#### Format
+The `scrollback` content can requested using timestamps:
+
+    SCROLLBACK target timestamp=YYYY-MM-DDThh:mm:ss.sssZ message_count
```

Comment:
> Wouldn't it be better to make it two parameters rather than using one separated with `=`?

No preference either way. Any reason two parameters are better?

> Can `message_count` be negative to go backwards?

Backwards in what sense? Send messages in reverse order or else? What is the use case?

 > Should the max message_count be in the ISUPPORT token?

`:irc.host 005 nick CHATHISTORY=message_count :are supported by this server`?

> What numerics are sent if message_count is invalid?

Oversight and not considered. Suggestions?